setup/updates/update-20230116.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+source ${SCRIPT_DIR}/../system.sh
+source ${SCRIPT_DIR}/../log.sh
+
+doil_update_20230116() {
+  doil_status_send_message_nl "Stopping all services"
+  doil_system_stop_all_services
+
+  doil_status_send_message "Removing old doil services"
+  doil_system_remove_services
+
+  doil_status_send_message_nl "Removing old doil system"
+  doil_system_remove_old_version
+  doil_status_okay
+
+  doil_status_send_message "Copying new doil system"
+  doil_system_create_folder
+  doil_system_copy_doil
+  doil_status_okay
+
+  doil_status_send_message "Run composer"
+  doil_system_run_composer
+  if [[ $? -ne 0 ]]
+  then
+    doil_status_failed
+    exit
+  fi
+  doil_status_okay
+
+  doil_status_send_message "Setting up access rights"
+  doil_system_setup_access
+  if [[ $? -ne 0 ]]
+  then
+    doil_status_failed
+    exit
+  fi
+  doil_status_okay
+
+  doil_status_send_message "Reinstalling salt service"
+  doil_system_replace_salt_stack
```

Comment:
This deletes _/usr/local/share/doil/setup/*_ and copies _/setup/stack/*_ to that directory a second time, after the copy part has already been done in _doil_system_copy_doil_ (without deleting files first). Wouldn't it be easier to include the deletion in _doil_system_copy_doil_ (if the directory exists) and only copy files once? This way we would also make sure that _doil_system_setup_access_ is done correctly for these files.
Or am I missing something essential here?

Reply:
Good hint. I place the deletion into doil_system_remove_old_version. Done!